### PR TITLE
Minor sts_session_token fixes aligning more closely to boto3 api

### PR DESCRIPTION
**Pull request by wimnat**
##### ISSUE TYPE
- Bugfix Pull Request
##### COMPONENT NAME

sts_session_token
##### ANSIBLE VERSION

2.2
##### SUMMARY

This module was merged quickly before I could review.

This PR just changes the return values to match boto3 as we want to keep values closely aligned to the API.  It also uses the helper function camel_dict_to_snake_dict to keep code DRY.

Also fixed the exception handling for AWS ClientError

### CODE REVIEW

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by wimnat** [comment]
Ping @pwnall

File: cloud/amazon/sts_session_token.py [hunk 1]
```diff
@@ -75,9 +75,9 @@
 
 # Use the session token obtained above to tag an instance in account 123456789012
 ec2_tag:
-  aws_access_key: "{{ session_credentials.sts_creds.access_key }}"
-  aws_secret_key: "{{ session_credentials.sts_creds.secret_key }}"
-  security_token: "{{ session_credentials.sts_creds.session_token }}"
+  aws_access_key: "{{ session_credentials.credentials.access_key_id }}"
```
Comment (pwnall) [inline cloud/amazon/sts_session_token.py:79]: This gets out of sync with https://github.com/ansible/ansible-modules-extras/blob/devel/cloud/amazon/sts_assume_role.py#L77

File: cloud/amazon/sts_session_token.py [hunk 2]
```diff
@@ -49,14 +49,14 @@
...
 
 RETURN = """
-sts_creds:
+credentials:
     description: The Credentials object returned by the AWS Security Token Service
     returned: always
     type: list
     sample:
-      access_key: ASXXXXXXXXXXXXXXXXXX
+      access_key_id: ASXXXXXXXXXXXXXXXXXX
```
Comment (pwnall) [inline cloud/amazon/sts_session_token.py:58]: This format was chosen for consistency with http://docs.ansible.com/ansible/sts_assume_role_module.html

Will you be changing that module too? I understand the appeal of aligning with the boto3 API, but I'm inclined to think that the train has already left the station when `sts_assume_role` was released.

File: cloud/amazon/sts_session_token.py [hunk 3]
```diff
@@ -122,18 +111,18 @@ def get_session_token(connection, module):
         response = connection.get_session_token(**args)
         changed = True
     except ClientError, e:
-        module.fail_json(msg=e)
+        module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
```
Comment (pwnall) [inline cloud/amazon/sts_session_token.py:128]: This hunk LGTM. Thank you for the simplification!

File: cloud/amazon/sts_session_token.py [hunk 4]
```diff
@@ -122,18 +111,18 @@ def get_session_token(connection, module):
...
-    module.exit_json(changed=changed, sts_creds=credentials)
 
 def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(
         dict(
-            duration_seconds = dict(required=False, default=None, type='int'),
-            mfa_serial_number = dict(required=False, default=None),
-            mfa_token = dict(required=False, default=None)
+            duration_seconds=dict(required=False, default=None, type='int'),
```
Comment (pwnall) [inline cloud/amazon/sts_session_token.py:137]: This hunk LGTM. Thank you for the extra type information!

**Comment by pwnall** [comment]
@wimnat can you please figure out the issue of being consistent with `sts_assume_role`?

I wish I could ping people to get this resolved, but I'm not sure who the API owners are.

**Comment by wimnat** [comment]
sts_assume_role uses boto and unfortunately was written prior to Ansible 2.0, when we got a bit clearer about how AWS modules should be written.

I don't think we should hold it up as an example of how to do things.  Let's make new modules clean and 'the right way'.  At some point, the Ansible core team will allow us to change older boto modules to boto3 (you can do that now but you need to maintain backward compat).  When the chance arises, the sts_assume_role module can be aligned.

**Comment by wimnat** [comment]
@pwnall any comments to push this along?

**Comment by pwnall** [comment]
@wimnat Sorry, for some reason I thought this was merged! Let me look through my inbox and understand what happened.

**Comment by gregdek** [comment]
Kicking this one to reset it.

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by wimnat** [comment]
@pwnall @gregdek @ryansb anything to do to get this in?

**Comment by gregdek** [comment]
Looks like this one never had a maintainer for some reason -- sorry for the mixup. Have added @pwnall as the owner in MAINTAINERS-EXTRAS, and kicking yet again.

@pwnall ball is in your court.

**Comment by gregdek** [comment]
Thanks @wimnat. To the current maintainers, @pwnall please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @wimnat for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@wimnat A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by wimnat** [comment]
ready_for_review

**Pull request closed without merging**